docs/patterns.md

```diff
@@ -142,3 +142,132 @@ Example access policy:
 * The front desk cannot cut releases nor deploy services
 
 Complex policies like this are currently best implemented in code directly, though there is [ongoing work](https://github.com/michaelansel/hubot-rbac) to build a generalized framework for access management.
+
+## Middleware Examples
+
+Middleware is often used for authentication and authorization.  There are three kinds of middleware:
```

Comment:
This feels like detail that should be in `scripting.md` (and maybe duplicated here). `patterns.md` seems like it should be more focused on use cases not on utilities (where middleware is a utility to implement use cases)